File: DSA Basics/Alogorithms/SEARCHING ALGORITHMS/Linear.py
# input from user
n = int(input("Enter the number of elements in the list: "))
# create an empty list
my_list = []
# loop to get the elements from user
for i in range(n):
    element = int(input(f"Enter element {i + 1}: "))
    my_list.append(element)

# input the target value to search for
target = int(input("Enter the target value to search for: "))

# linear search algorithm
def linear_search(lst, target):
    for index in range(len(lst)):
        if lst[index] == target:
            return index  # return the index if target is found
    return -1  # return -1 if target is not found

# call the linear search function
result = linear_search(my_list, target)
# output the result
if result != -1:
    print(f"Target value {target} found at index {result}.")
else:
    print(f"Target value {target} not found in the list.")

File: DSA Basics/Alogorithms/SEARCHING ALGORITHMS/Linear.java
public class Linear{
    public static int  linearSearch(int number[],int key){
        for(int i=0;i<number.length;i++){
            if(number[i]==key){
                return i;
            }
        }
        return -1;
    }
    public static void main(String[] args) {
        int numbers[] = {2,4,6,8,10,12,14,16};
        int key = 6;
        int index = linearSearch(numbers, key);
        if(index == -1){
            System.out.println("key not found");
        }else{
        System.out.println("key is found at index :"+index);
        }
    }
}